lib/frontman/context.rb

```diff
@@ -38,28 +38,25 @@ def wrap_layout(layout)
         .returns(T.untyped)
     end
     def content_for(key, content = nil)
-      # Haml is not designed to do handle content_for properly
-      # so we need to hack the buffer of haml that is set
-      # inside the context by haml
-      save_buffer
+      set_content_buffer(content)
```

Comment:
Shouldn't we use the `content` that the `set_content_buffer` method returns?

```suggestion
      content = set_content_buffer(content)
```

I think we also have to forward any blocks given to the `content_for` method to this new method, because it uses this to yield any content.

Reply:
Fixed.